R/create_obi_cohort.R

```diff
@@ -11,155 +11,281 @@
+          mdhhs_id %in% c(
+            "25006",
+            "63019",
+            "63029",
+            "82053",
+            "13013",
+            "63023",
+            "82514",
+            "63018",
+            "82537",
+            "82535",
+            "63015",
+            "82058",
+            "82024",
+            "38009",
+            "50014",
+            "70001",
+            "33002",
+            "74004",
+            "81006",
+            "28002",
+            "33001"
+          ),
+          1,
+          0
         )
-      ),
-      g1_site_2023 = ifelse(
-        mdhhs_id %in% c(
-          "25006",
-          "82053",
-          "39002",
-          "82514",
-          "63015",
-          "73008",
-          "82058",
-          "82028",
-          "63031",
-          "70001",
-          "25005",
-          "56002",
-          "33001",
-          "63009",
-          "81005",
-          "41010",
-          "41005"
+      )
+    
+    if (limit_to_locked == T) {
+      df1 |> filter(
+        flg_complete == 1,
+        case_locked == 1,
+        infant_dob_dt >= lubridate::ymd_hms("2020-01-01 00:00:00")
+      )
+    }
+    
+    else if (limit_to_locked == F) {
+      df1 |> filter(flg_complete == 1,
+                    infant_dob_dt >= lubridate::ymd_hms("2020-01-01 00:00:00"))
+    }
+    
+  } else{
+    df1 = df |>
+      mutate(
+        infant_dob_dt_hms = lubridate::ymd_hms(infant_dob_dt),
+        infant_dob_dt = lubridate::as_date(infant_dob_dt_hms),
+        #case locks at MIDNIGHT AFTER THIS DATE
+        case_lock_dt = infant_dob_dt_hms + days(91),
+        case_locked = ifelse(as_date(case_lock_dt) < lubridate::today(), 1, 0),
+        across(
+          c(
+            starts_with("opioid_e"),
+            starts_with("opioid_dose"),
+            starts_with("opioid_quantity_no"),
+            starts_with("opioid_unit")
+          ),
+          ~ ifelse(discharge_opioid_e == 1, .x, NA)
         ),
-        1,
-        0
-      ),
-      g1_color_2023 = case_when(
-        mdhhs_id %in% c(
-          "39002",
-          "63009",
-          "41010"
-        ) ~ "green",
-        mdhhs_id %in% c(
-          "82514",
-          "63031",
-          "81005"
-        ) ~ "yellow",
-        mdhhs_id %in% c(
-          "25006",
-          "82053",
-          "63015",
-          "73008",
-          "82058",
-          "82028",
-          "70001",
-          "25005",
-          "56002",
-          "33001",
-          "41005"
-        ) ~ "red",
-        TRUE ~ as.character(NA)
-      ),
-      select_g1_site_2023 = ifelse(
-        mdhhs_id %in% c(
-          "25006",
-          "82053",
-          "63015",
-          "73008",
-          "82058",
-          "82028",
-          "70001",
-          "25005",
-          "56002",
-          "33001",
-          "41005"
+        mdhhs_id = as.character(external_mdhhs_site_id),
+        mdhhs_id = ifelse(nchar(mdhhs_id) < 5, paste0("0", mdhhs_id), mdhhs_id),

```

Comment:
external_mdhhs_site_id is specified as 5 digits character with 0 as padding when needed. So, you can just use external_mdhhs_site_id. 

so we don't lead to any confusion of what the difference between external_mdhhs_site_id and mdhhs_id in the dataset